tests/src/tests.rs

```diff
@@ -30,10 +29,64 @@ const PDF_DIR: &str = "pdf";
 const FONT_DIR: &str = "../assets/fonts";
 const FILE_DIR: &str = "../assets/files";
 
+use clap::{ArgAction::*, Parser};
+
+#[derive(Debug, Clone, Parser)]
+#[clap(name = "typst-test", author)]
+struct Args {
+    filter: Vec<String>,
+    // runs only the subtest specified
+    #[arg(short, long)]
+    subtest: Option<usize>,
+    #[arg(long, action = SetTrue)]
+    exact: bool,
+    #[arg(long, action = SetTrue)]
+    pdf: bool,
+    #[arg(long, action = SetTrue)]
+    syntax: bool,
+    #[arg(long, action = SetTrue)]
+    model: bool,
+    #[arg(long, action = SetTrue)]
+    frame: bool,
+    #[arg(long, action = Count)]
+    nocapture: u8, // simply ignores the argument
+}
+
+/// Which things to print out for debugging.
+#[derive(Default, Debug, Copy, Clone, Eq, PartialEq)]
+struct PrintConfig {
+    syntax: bool,
+    model: bool,
+    frames: bool,
+}
+
+impl PrintConfig {
+    fn from_arg(arg: &Args) -> Self {
+        Self {
+            syntax: arg.syntax,
+            model: arg.model,
+            frames: arg.frame,
+        }
+    }
+}
```

Comment:
I guess we can get rid of this and just use `Args` in its place. Or maybe have the clap stuff on here if there is an equivalent to `serde(flatten)` for clap.